haproxy_discovery.sh

```diff
@@ -14,8 +14,24 @@
 HAPROXY_SOCK="/var/run/haproxy/info.sock"
 [ -n "$1" ] && echo $1 | grep -q ^/ && HAPROXY_SOCK="$(echo $1 | tr -d '\040\011\012\015')"
 
+if [[ "$1" =~ (25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?):[0-9]{1,5} ]];
```

Comment:
well, I'd say the socket should then be specified like `file:///var/run/...`, so both argument types are URIs, but again for backwards compatibility that should probably not be touched.